### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by darinwil** [longhorn/longhorn#4231]
## Describe the bug

Adding RWX volumes lead to ContainerWaiting status and an error:

MountVolume.MountDevice failed for volume "pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f" : rpc error: code = Internal desc = mount failed: exit status 32 Mounting command: /usr/local/sbin/nsmounter Mounting arguments: mount -t nfs -o vers=4.1,noresvport,soft,intr,timeo=30,retrans=3 10.43.39.233:/pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f/globalmount Output: mount.nfs: No such device

## To Reproduce

1. Install PhotonOS
2. Install K3S
3. Install Rancher
4. Install Longhorn Prerequisites
5. Install Longhorn
## Expected behavior

RWX volumes mount properly

## Log or Support bundle

MountVolume.MountDevice failed for volume "pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f" : rpc error: code = Internal desc = mount failed: exit status 32 Mounting command: /usr/local/sbin/nsmounter Mounting arguments: mount -t nfs -o vers=4.1,noresvport,soft,intr,timeo=30,retrans=3 10.43.39.233:/pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-f2ab97f6-b965-41be-a1e2-6095ba1a458f/globalmount Output: mount.nfs: No such device

## Environment

 - Longhorn version: v1.2.4
 - Installation method Rancher Catalog App
 - Kubernetes distro K3s
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 5
 - Node config
   - OS type and version: PhotonOS 4
   - CPU per node: 8
   - Memory per node: 15Gi
   - Disk type(e.g. SSD/NVMe): VMWare vSAN
   - Network bandwidth between the nodes: 10G
 - Underlying Infrastructure: VMWare
 - Number of Longhorn volumes in the cluster: 13

**Comment by darinwil** [comment]
[longhorn-support-bundle_cf907e2f-5ef4-461c-be48-50b4c9acc7d9_2022-07-13T12-28-55Z.zip](https://github.com/longhorn/longhorn/files/9102088/longhorn-support-bundle_cf907e2f-5ef4-461c-be48-50b4c9acc7d9_2022-07-13T12-28-55Z.zip)

**Comment by mantissahz** [comment]
Hi @darinwil Could you please check the `nfs-utils` installed and does it support nfs v4?

**Comment by darinwil** [comment]
It is and it is supposed to, I have nfs-utils-2.5.1-3.ph4.x86_64 installed and there is /sbin/mount.nfs4 . I also noticed that rpcbind was not running on all the nodes, I started and enabled it hoping that was the problem but deleting and recreating the deployment with the RWX volume still fails.

**Comment by darinwil** [comment]
Okay I finally figured this out, apparently when the kernel is updated on PhotonOS it deletes the currently running kernel modules directories or something like that. I found on one of the nodes that the nfsd module was not loaded and when I tried to load it modprobe complained that it couldn't find a path, which I was able to find in the staged updated kernel module path. Rebooting each node (ensuring that rpcbind was enabled in systemd) has corrected the issue. To prevent it in the future I will add an exclude to the tdnf.conf file excluding the (in my case linux-esx) package and manually updating the kernel. Thanks for the help!
